apis/fluent-en_GB/extensions/jdk8/atrium-api-fluent-en_GB-jdk8-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/jdk8/pathAssertions.kt

```diff
@@ -55,3 +56,28 @@ val <T : Path> Expect<T>.parent get(): Expect<Path> = ExpectImpl.path.parent(thi
  */
 fun <T : Path> Expect<T>.parent(assertionCreator: Expect<Path>.() -> Unit): Expect<T> =
     ExpectImpl.path.parent(this).addToInitial(assertionCreator)
+
+/**
+ * Expects that this [Path] has a [fileNameWithoutExtension][Path.fileNameWithoutExtension] and creates an [Expect] for it,
```

Comment:
This does not reflect 100% the current behaviour. Your assertion function does not check if Path has an extension.
I suggest you make this explicit by adding a test case for `path without extension`, for instance `Paths.get("a/")` and change the KDoc to my suggestion:
```
/**
 * Creates an [Expect] for the property [Path.fileNameWithoutExtension]
 * (provided via [niok](https://github.com/robstoll/niok)) of the subject of the assertion,
 * so that further fluent calls are assertions about it.
 *
 * @return This assertion container to support a fluent API.
 * @throws AssertionError Might throw an [AssertionError] if the assertion made is not correct.
 *
 * @since 0.9.0
 */
```

Reply:
Changed and added two more test cases:
- file without extension
- directory path